pkg/provisioner/ironic/ironic.go

```diff
@@ -203,6 +209,12 @@ func (p *ironicProvisioner) findExistingHost() (ironicNode *nodes.Node, err erro
 	}
 }
 
+// BootMode returns the boot method to be used for the host, based on
+// the BMC driver.
+func (p *ironicProvisioner) BootMode() v1alpha1.BootMode {
+	return p.bmcAccess.DefaultBootMode()
```

Comment:
Do we need some way of getting the *allowed* boot modes?

Reply:
What would that be used for?

Reply:
So we could ignore the user's preference if they require something Ironic can't support.
But maybe we just go ahead and try to provision and they get an error if they did it wrong.

Reply:
Well, provision should fail... that is unless the BMC is completely disjointed from reality. Unfortunately that is a thing, yet extremely rare and vendors I'm aware of having started to head down that path quickly reverted once people started complaining.

Realistically, if we're using IPMI, there is just no solid and consistent way to know. Some vendors are only settable via raw flags, others are only settable via custom boot target. Luckily everyone seems to support the raw flags, as far as I'm aware.

Reply:
> Well, provision should fail... that is unless the BMC is completely disjointed from reality. Unfortunately that is a thing, yet extremely rare and vendors I'm aware of having started to head down that path quickly reverted once people started complaining.
> 
> Realistically, if we're using IPMI, there is just no solid and consistent way to know. Some vendors are only settable via raw flags, others are only settable via custom boot target. Luckily everyone seems to support the raw flags, as far as I'm aware.

We've established that there is no way for us to know for sure, that's why we have the new field.

The only question in my mind is how to pick a default. Should we ask the BMC? Or should we use a single default all the time?

Reply:
Sounds good to me.